src/rdkafka_telemetry.c

```diff
@@ -343,20 +343,22 @@ static void rd_kafka_send_push_telemetry(rd_kafka_t *rk,
                                          rd_bool_t terminating) {
 
         rd_buf_t *metrics_payload = rd_kafka_telemetry_encode_metrics(rk);
-        size_t compressed_metrics_payload_size = 0;
-        void *compressed_metrics_payload       = NULL;
-        rd_kafka_compression_t compression_used =
-            rd_kafka_push_telemetry_payload_compress(
-                rk, rkb, metrics_payload, &compressed_metrics_payload,
-                &compressed_metrics_payload_size);
-        if (compressed_metrics_payload_size >
-            (size_t)rk->rk_telemetry.telemetry_max_bytes) {
-                rd_kafka_log(rk, LOG_WARNING, "TELEMETRY",
-                             "Metrics payload size %" PRIusz
-                             " exceeds telemetry_max_bytes %" PRId32
-                             "specified by the broker.",
-                             compressed_metrics_payload_size,
-                             rk->rk_telemetry.telemetry_max_bytes);
+        size_t compressed_metrics_payload_size  = 0;
+        void *compressed_metrics_payload        = NULL;
+        rd_kafka_compression_t compression_used = RD_KAFKA_COMPRESSION_NONE;
+        if (metrics_payload) {
+                compression_used = rd_kafka_push_telemetry_payload_compress(
+                    rk, rkb, metrics_payload, &compressed_metrics_payload,
+                    &compressed_metrics_payload_size);
+                if (compressed_metrics_payload_size >
+                    (size_t)rk->rk_telemetry.telemetry_max_bytes) {
+                        rd_kafka_log(rk, LOG_WARNING, "TELEMETRY",
+                                     "Metrics payload size %" PRIusz
+                                     " exceeds telemetry_max_bytes %" PRId32
+                                     "specified by the broker.",
+                                     compressed_metrics_payload_size,
+                                     rk->rk_telemetry.telemetry_max_bytes);
+                }
         }
```

Comment:
maybe we can also add a debug log mentioning we're sending an empty metrics blob